File: libs/power-log-parser/src/lib/handlers/show-entity-handler.ts
import type { Helper } from '../helper';
import { Action, Game, Node, NodeType, ShowEntity, Tag } from '../models';
import { Regexes } from '../regexes';
import type { ParserState } from '../state/parser-state';

export class ShowEntityHandler {
	static Handle(
		timestamp: string,
		data: string,
		state: ParserState,
		indentLevel: number,
		helper: Helper,
	): boolean {
		const match = Regexes.ActionShowEntityRegex.exec(data);
		if (match) {
			const rawEntity = match[1];
			const cardId = match[2];
			const entity = helper.ParseEntity(rawEntity);

			const showEntity = new ShowEntity();
			showEntity.CardId = cardId;
			showEntity.Entity = entity;
			showEntity.Tags = [];
			showEntity.TimeStamp = timestamp;
			showEntity.SubSpellInEffect = state.CurrentSubSpell?.GetActiveSubSpell() ?? null;

			state.UpdateCurrentNode(NodeType.Game, NodeType.Action);
			if (state.Node!.Type === NodeType.Game) {
				(state.Node!.Object as Game).AddData(showEntity);
			} else if (state.Node!.Type === NodeType.Action) {
				(state.Node!.Object as Action).Data.push(showEntity);
			} else {
				throw new Error('Invalid node ' + state.Node!.Type + ' while parsing ' + data);
			}
			const newNode = new Node(NodeType.ShowEntity, showEntity, indentLevel, state.Node, data);
			state.CreateNewNode(newNode);
			state.Node = newNode;
			return true;
		}
		return false;
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Parser/Handlers/DataHandler/ShowEntityHandler.cs
using HearthstoneReplays.Enums;
using HearthstoneReplays.Parser.ReplayData.Entities;
using HearthstoneReplays.Parser.ReplayData.GameActions;
using HearthstoneReplays.Parser.ReplayData;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Action = HearthstoneReplays.Parser.ReplayData.GameActions.Action;

namespace HearthstoneReplays.Parser.Handlers
{
    internal class ShowEntityHandler
    {
        public static bool HandleShowEntity(DateTime timestamp, string data, ParserState state, int indentLevel, Helper helper)
        {
            var match = Regexes.ActionShowEntityRegex.Match(data);
            if (match.Success)
            {
                var rawEntity = match.Groups[1].Value;
                var cardId = match.Groups[2].Value;
                var entity = helper.ParseEntity(rawEntity);

                var showEntity = new ShowEntity { CardId = cardId, Entity = entity, Tags = new List<Tag>(), TimeStamp = timestamp };
                showEntity.SubSpellInEffect = state.CurrentSubSpell?.GetActiveSubSpell();
                state.UpdateCurrentNode(typeof(Game), typeof(Action));
                if (state.Node.Type == typeof(Game))
                    ((Game)state.Node.Object).AddData(showEntity);
                else if (state.Node.Type == typeof(Action))
                    ((Action)state.Node.Object).Data.Add(showEntity);
                else
                    throw new Exception("Invalid node " + state.Node.Type + " while parsing " + data);
                var newNode = new Node(typeof(ShowEntity), showEntity, indentLevel, state.Node, data);
                state.CreateNewNode(newNode);
                state.Node = newNode;
                return true;
            }
            return false;
        }
    }
}
